### PR TITLE
Add support for testing whether a value is a camel-cased string

### LINKED ISSUE #533: [RFC]: add support for testing whether a value is a camel-cased string
### Description

This RFC proposes to add support for testing whether a value is a camel-cased string.

```javascript
var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
var camelcase = require( '@stdlib/string/base/camelcase' );

function isCamelcase( value ) {
    return ( isString( value ) && camelcase( value ) === value );
}
```

Package: `@stdlib/assert/is-camelcase`
alias: `isCamelcase`

### Related Issues

No.

### Questions

No.

### Other

Reference packages:

-   [`@stdlib/assert/is-uppercase`](https://github.com/stdlib-js/stdlib/tree/13c1fa39ff73208d819697dc07917bdb7539f295/lib/node_modules/%40stdlib/assert/is-uppercase)

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #533 .

## Checklist

- [x] Add lib
- [x] Add examples
- [x] Add package.json
- [x] Add benchmark
- [x] Add etc
- [x] Add bin
- [x] Add readme.md
- [x] Add test
- [x] Add docs

* * *
@kgryte 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Pranavchiku** [comment]
Some tests for ```test.cli.js``` is failing and ```docs``` is giving me ```typescript``` errors (maybe I performed git clean and hence few .json files are absent) if you could check out once, and will try to figure out, once resolved will push the remaining changes :)

**Comment by Pranavchiku** [comment]
We need ```test/fixtures``` for this PR right?

**Comment by kgryte** [comment]
@Pranavchiku Correct. As in `@stdlib/assert/is-uppercase`, you need a fixture file for running the CLI tests which test `stdin`. I recommend just following the precedent set forth in that package.

File: lib/node_modules/@stdlib/assert/is-camelcase/bin/cli [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env node
+
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-camelcase/bin/cli]:
```suggestion
* Copyright (c) 2022 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/assert/is-camelcase/lib/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Test if a string is in camelcase.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-camelcase/lib/index.js]:
```suggestion
* Test if a value is a camelcase string.
```

File: lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var camelcase = require( '@stdlib/string/base/camelcase' );
+
+
+// MAIN //
+
+/**
+* Test if a string is in camelcase.
+* @param {*} value - value to test
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js:32]:
```suggestion
*
* @param {*} value - value to test
```

File: lib/node_modules/@stdlib/assert/is-camelcase/package.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+  },
+  "contributors": [
+    {
+      "name": "The Stdlib Authors",
+      "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+    }
+  ],
+  "bin": {
+    "is-uppercase": "./bin/cli"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-camelcase/package.json]:
```suggestion
    "is-camelcase": "./bin/cli"
```

**Comment by kgryte** [comment]
This PR is looking good. Just missing the TypeScript declarations and the CLI tests. Thanks for working on this!

**Comment by kgryte** [comment]
For the TypeScript declarations, hard to say what is happening. I'd just copy-and-paste the declarations from `@stdlib/assert/is-uppercase` and modify accordingly. If the declaration tests pass for that package, they should pass for this package.

**Comment by Pranavchiku** [comment]
Yeah I will do that changes and push those :)

**Comment by Pranavchiku** [comment]
@kgryte this PR can have a final review :))

File: lib/node_modules/@stdlib/assert/is-camelcase/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.9
+
+/**
+* Test if a value is a camelcase string.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-camelcase/docs/types/index.d.ts]:
```suggestion
* Tests if a value is a camelcase string.
```

File: lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var camelcase = require( '@stdlib/string/base/camelcase' );
+
+
+// MAIN //
+
+/**
+* Test if a string is in camelcase.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js]:
```suggestion
* Tests if a string is in camelcase.
```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js]: I have to covert "Tests" to "Test" else ```eslint``` gives me error on committing the changes

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

File: lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js (hunk 6 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/assert/is-camelcase/lib/main.js]: @Planeshifter Did we possibly change something in the lint rules?

**Pull request merged**